x-pack/filebeat/module/microsoft/defender_atp/test/defender_atp-test.json.log-expected.json

```diff
@@ -0,0 +1,256 @@
+[
+    {
+        "@timestamp": "2020-06-30T10:09:01.1569718Z",
+        "cloud.account.id": "123543-d66c-4c7e-9e30-40034eb7c6f3",
+        "cloud.instance.id": "c5a964f417c11f6277d5bf9489f0d",
+        "cloud.provider": "azure",
+        "event.action": "Malware",
+        "event.category": [
+            "host",
+            "malware"
+        ],
+        "event.dataset": "microsoft.defender_atp",
+        "event.duration": 0,
+        "event.end": "2020-06-30T10:07:44.333733Z",
+        "event.id": "da637291085411733957_-1043898914",
+        "event.ingested": "2020-07-06T21:45:21.199464Z",
+        "event.kind": "alert",
+        "event.module": "microsoft",
+        "event.provider": "defender_atp",
+        "event.severity": 2,
+        "event.start": "2020-06-30T10:07:44.333733Z",
+        "event.timezone": "UTC",
+        "event.type": [
+            "end"
+        ],
+        "file.hash.sha1": null,
```

Comment:
the rename processor is renaming when the value is null, I'm wondering if we should switch to convert or set processor.  This applies to several of the fields.

Reply:
Ah that is a good point. Since ATP uses a Odata Schema, the full object is always returned, so that has to be performed on all the defender_atp fields (about 15), will add that in today!

Reply:
I resolved this by ensuring that all null fields are stripped with a script at the top. It seems to be something we could reuse for other modules as well, that way we can still use "ignore_missing" for renames.